resources/js/Organisms/Form.tsx

```diff
@@ -48,15 +49,49 @@ const Form: React.FC<IFormProps> = ({ action, initialValues, render }) => {
   };
 
   const updateState = (target: HTMLInputElement) => {
-    const { name, value } = target;
-    setValues({ ...values, [name]: value });
+    const { name, value,type } = target;
+    if(type==='checkbox'){
+      let newSelection: any[];
+      if(values[name].indexOf(value) > -1) 
+      {
+        newSelection = values[name].filter(s => s !== value)
+      }else 
+      {
+        newSelection = [...values[name], value];
+      }
+      setValues({ ...values, [name]: newSelection }); 
+    }else
+      {
+        setValues({ ...values, [name]: value });
+      }
+
+    if(haveErrors(errors))
+      {
+        validateForm();
+      }
   };
 
   const validateForm = (): boolean => {
-    // TODO - validate form
-    // (set errors? for now just set as empty object)
-    setErrors({});
-    return true;
+    let newErrors:IErrors = errors;
+    let formisValid:boolean = true;
+    newErrors.fullname ='';
+    newErrors.email ='';
+
+    if (!values["fullname"] || values["fullname"].trim().length==0) 
+      {
+        newErrors.fullname ='Please enter your fullname';
+        formisValid=false;
+      }
+
+    let validEmail = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/; 
```

Comment:
I think it's not sensible to try and validate email addresses so strictly, because we risk accidentally blocking someone from signing up with an unusual, but valid email address. If you're interested, [here](https://davidcel.is/posts/stop-validating-email-addresses-with-regex/) is a blog post with a few more details about this subject.

I think we should make this check a lot simpler, something like this: (check that that they typed some character(s), an @ sign, followed by some more character(s) with a dot in there somewhere):
```suggestion
    let validEmail = /^.+@.+\..+$/; 
```

I appreciate that not everyone agrees about this topic, so feel free to leave this as is if you prefer 😄